src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -512,19 +515,25 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             {
                 sendTask = base.SendAsync(request, cts.Token);
             }
+            catch (OperationCanceledException e) when (TimeoutFired(cancellationToken, timeoutTime))
+            {
+                HandleFinishSendAsyncCleanup(cts, disposeCts);
+                ThrowTimeoutException(e);
+                throw;
+            }
```

Comment:
I still think it makes more sense for this to be:
```C#
throw CreateTimeoutException(e);
```
rather than:
```C#
ThrowTimeoutException(e);
throw;
```

Reply:
Done